Jsonparser.cpp

```diff
@@ -0,0 +1,194 @@
+#include "Jsonparser.h"
+
+Jsonparser::Jsonparser(std::ifstream & f) {
+	int keyv, i;
+	std::string::size_type found;
+	if (!f.good()) throw 56;
+	else {
+		std::string sor = "";
+		std::string tmp;
+
+		while (!f.eof()) {
+			getline(f, sor);
+			keyv = 0;
+			found = sor.find('"');
+			if (found != std::string::npos) {
+				i = found + 1;
+				tmp = "";
+				while (sor[i] != '"') {
+					tmp = tmp + sor[i];
+					i++;
+				}
+
+				if (tmp == "name") keyv = 1;
+				if (tmp == "hp") keyv = 2;
+				if (tmp == "dmg") keyv = 3;
+			}
+
+			found = sor.find(':');
+			if (found != std::string::npos) {
+				i = found + 1;
+				tmp = "";
+				while (sor[i] != ',' && i != sor.size()) {
+					if (sor[i] == '"' || sor[i] == ' ') {
+						i++;
+					}
+					else {
+						tmp = tmp + sor[i];
+						i++;
+					}
+				}
+
+				switch (keyv) {
+				case 1: m["name"] = tmp;
+					break;
+				case 2: m["hp"] = tmp;
+					break;
+				case 3: m["dmg"] = tmp;
+					break;
+				default: throw 57;
+					break;
+
+				}
```

Comment:
Mi lenne, ha nem akarnád ugyanabban a tmp változóban tárolni a kulcsot és az értéket is? A kulcs lehetne pl. a key-ben, az érték meg value-ban. Utána el lehetne intézni mindent egy m[key] = value-val, nem kell keyv és néhány művelet, amit azzal végzel.

Reply:
A megengedett kulcs értékeket meg fel lehet venni egy tömbbe, amin végig lehet iterálni. Ha nincs köztük a kulcs, dobhatsz kivételt.

Reply:
Vagy halmazba, az talán még elegánsabb.